### ISSUE TITLE
[Bug] Azure Execute hangs for ~25 minutes and then fails with generic node error.

### ISSUE THREAD

**Issue by scphantm** [GitTools/GitVersion#2864]
**Describe the bug**
i reported this here https://github.com/GitTools/actions/issues/511 but havn't gotten any response, my logs and such are over there. 

Heres the basic, when we are seeding a repo, the azure pipeline will hang on the GitVersion Execute task.  we don't know why, but when it does happen, it will hang for about 30 minutes and then fail with a generic node error.  What fixes it is simply creating a new PR with just a white space commit to it.

the bug is 2 fold

1 - why is it failing
2 - why is it failing in such a way that we can't debug.

## Expected Behavior
It works

## Actual Behavior
It hangs on the execute task until Node or Azure times out.

## Possible Fix
putting in a white space commit makes it work again.

## Steps to Reproduce
when we are seeding our repos, the initial commits cause the thing to hang.  Simply adding a blank white space commit will clear the error.

## Context
we have hundreds of repos and pipelines.  This is killing us.

## Your Environment
Azure Pipelines

- Version Used:
- Operating System and version (Windows 10, Ubuntu 18.04):
- Link to your project:
- Link to your CI build (if appropriate):

**Comment by scphantm** [comment]
I just posted this in the other bug:

we are really having a problem with this.  its bad enough where im getting pressure to pull the tool.  we show about 15% of our pipelines fail because of this error.  The problem case has expanded to include pipelines that have already worked correctly.  meaning we did the empty commit mentioned above and got it to work, then the next PR, it fails.

Any ideas here?  It really seems like node is either hanging because it doesn't like something in the repo, or, what i really think is happening is gitversion is failing for some reason and node isn't picking up the exit code or logs.  possibly because the action has it always parsing json?  if there is an error, is the return 

https://github.com/GitTools/actions/blob/d957a15d75c7fc44255d91da910f7a2c0b0be42f/src/tasks/gitversion/main.ts#L49

here, able to capture and return an error?  I know in testing locally, if gitversion generates an error, the output to the screen is not json.  Its only json when things go correctly.
